src/app/common/hooks/use-convert-to-fiat-amount.ts

```diff
@@ -2,29 +2,64 @@ import { useCallback, useMemo } from 'react';
 
 import { CryptoCurrencies } from '@shared/models/currencies.model';
 import { createMarketData, createMarketPair } from '@shared/models/market.model';
-import type { Money } from '@shared/models/money.model';
+import { type Money, createMoney } from '@shared/models/money.model';
+import { isUndefined } from '@shared/utils';
 
-import { useCryptoCurrencyMarketData } from '@app/query/common/market-data/market-data.hooks';
+import { i18nFormatCurrency } from '@app/common/money/format-money';
+import { unitToFractionalUnit } from '@app/common/money/unit-conversion';
+import { useCryptoCurrencyMarketDataMean } from '@app/query/common/market-data/market-data.hooks';
 
 import { baseCurrencyAmountInQuote } from '../money/calculate-money';
 
-export function useConvertCryptoCurrencyToFiatAmount(currency: CryptoCurrencies) {
-  const cryptoCurrencyMarketData = useCryptoCurrencyMarketData(currency);
+export function useConvertCryptoCurrencyToFiatAmountMean(currency: CryptoCurrencies) {
+  const cryptoCurrencyMarketData = useCryptoCurrencyMarketDataMean(currency);
 
   return useCallback(
     (value: Money) => baseCurrencyAmountInQuote(value, cryptoCurrencyMarketData),
     [cryptoCurrencyMarketData]
   );
 }
 
-export function useConvertAlexSdkCurrencyToFiatAmount(currency: CryptoCurrencies, price: Money) {
-  const alexCurrencyMarketData = useMemo(
+function useConvertCryptoCurrencyToFiatAmount(currency: CryptoCurrencies, price: Money) {
+  const cryptoCurrencyMarketData = useMemo(
     () => createMarketData(createMarketPair(currency, 'USD'), price),
     [currency, price]
   );
 
   return useCallback(
-    (value: Money) => baseCurrencyAmountInQuote(value, alexCurrencyMarketData),
-    [alexCurrencyMarketData]
+    (value: Money) => baseCurrencyAmountInQuote(value, cryptoCurrencyMarketData),
+    [cryptoCurrencyMarketData]
+  );
+}
+
+export function useFormattedFractionalUnitAsFiat(balance?: Money, price?: Money, value?: string) {
```

Comment:
Why does this function have optional params?

The fact these values can sometimes be `undefined` (APIs being down, not loaded yet etc) is not the concern of this function performing mathematical operations.

Reply:
I feel like this was setup prior to me using the market data code (market pairs, etc) and uses quite a lot of hooks that can't be conditional? It does all seem complicated, but I didn't write initially and would take some time to refactor all of it. I'll double check, but I think I had a hard time working with all the hooks.